apps/customer-portal/backend-v2/cmd/server/main.go

```
@@ -367,6 +369,45 @@ func main() {
+	//  2. The REST server's Read/WriteTimeout would cap the lifetime of an
+	//     upgraded connection. The handler clears those deadlines after the
+	//     upgrade, but only a listener without them is safe by construction.
+	//
+	// Each Choreo endpoint maps to one port, so this also gives the
+	// customer-portal-websocket endpoint a port of its own — see
+	// .choreo/component.yaml.
+	wsMux := http.NewServeMux()
+	wsMux.HandleFunc("GET /ws", webSocketHandler.HandleWebSocket)
+
+	wsAddr := ":" + mustPort("WS_PORT", "8081")
+	wsLn, err := net.Listen("tcp", wsAddr)
```

golangci-lint (2.12.2)

[error] net.Listen must not be called. use (*net.ListenConfig).Listen

(noctx)